src/utils/data_loaders.py

```diff
@@ -13,157 +17,135 @@ def load_tsv(
-            row = row.strip().split(delimiter)
-            if filter_idx and row[filter_idx] != filter_value:
-                continue
-            sent1 = process_sentence(tokenizer_name, row[s1_idx], max_seq_len)
-            if (targ_idx is not None and not row[targ_idx]) or not len(sent1):
-                continue
-
-            if targ_idx is not None:
-                if targ_map is not None:
-                    targ = targ_map[row[targ_idx]]
-                elif targ_fn is not None:
-                    targ = targ_fn(row[targ_idx])
-                else:
-                    targ = int(row[targ_idx])
-            else:
-                targ = 0
-
-            if s2_idx is not None:
-                sent2 = process_sentence(tokenizer_name, row[s2_idx], max_seq_len)
-                if not len(sent2):
-                    continue
-                sent2s.append(sent2)
-
-            if idx_idx is not None:
-                idx = int(row[idx_idx])
-                idxs.append(idx)
-
-            sent1s.append(sent1)
-            targs.append(targ)
-
-    if idx_idx is not None:
-        return sent1s, sent2s, targs, idxs
+        has_labels=True,
+        filter_value=None):
+    '''
+    Load a tsv.
+    To load only rows that have a certain value for a certain column,
+    like genre in MNLI, set filter_idx and filter_value (for example,
+    for mnli-fiction  we want columns where genre == 'fiction' ).
+    Args:
+        s1_idx; int
+        s2_idx: int
+        targ_idx: int
+        filter_idx: int this is the index that we want to filter from
+        filter_value: string the value in which we want filter_idx to be equal to
+        return_indices: bool that describes if you need to return indices (for purposes of matching)
+        label_fn is a function that expects a row and outputs the label
+    Returns:
+        List of first and second sentences, labels, and if applicable indices
+    '''
+    # TODO(Yada): Instead of index integers, adjust this to pass ins column names
+    # get the first row as the columns to pass into the pandas reader
+    # This reads the data file given the delimiter, skipping over any rows (usually header row)
+    rows = pd.read_csv(data_file, \
+                        sep=delimiter, \
+                        error_bad_lines=False, \
+                        names=col_indices, \
+                        header=None, \
+                        skiprows=skip_rows, \
+                        quoting=csv.QUOTE_NONE,\
+                        encoding='utf-8')
+    if filter_idx:
+        rows = rows[rows[filter_idx] == filter_value]
+    sent2s = pd.Series()
+    # Filter for sentence1s that are of length 0
+    # Filter if row[targ_idx] is nan
+    mask = (rows[s1_idx].str.len() > 0)
+    if has_labels:
+        mask = mask & rows[label_idx].notnull()
+    rows = rows.loc[mask]
+    sent1s = rows[s1_idx].apply(lambda x: process_sentence(tokenizer_name, x, max_seq_len))
+    if s2_idx is not None:
```

Comment:
nit: have both cases here rather than relying on the default init earlier:
```
if s2_idx is None:
    sent2s = pd.Series()  # or should this be None?
else:
    sent2s = rows[s2_idx].apply(...)
```